src/sentry/integrations/pagerduty/actions/form.py

```diff
@@ -29,28 +30,26 @@ class PagerDutyNotifyServiceForm(forms.Form):
     account = forms.ChoiceField(choices=(), widget=forms.Select())
     service = forms.ChoiceField(required=False, choices=(), widget=forms.Select())
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         integrations = [(i.id, i.name) for i in kwargs.pop("integrations")]
         services = kwargs.pop("services")
 
         super().__init__(*args, **kwargs)
         if integrations:
             self.fields["account"].initial = integrations[0][0]
 
-        self.fields["account"].choices = integrations
-        self.fields["account"].widget.choices = self.fields["account"].choices
+        set_field_choices(self.fields["account"], integrations)
 
         if services:
             self.fields["service"].initial = services[0][0]
 
-        self.fields["service"].choices = services
-        self.fields["service"].widget.choices = self.fields["service"].choices
+        set_field_choices(self.fields["service"], services)
 
     def _validate_service(self, service_id: int, integration_id: int) -> None:
         with record_event(OnCallInteractionType.VALIDATE_SERVICE).capture() as lifecycle:
             params = {
-                "account": dict(self.fields["account"].choices).get(integration_id),
-                "service": dict(self.fields["service"].choices).get(service_id),
+                "account": dict(get_field_choices(self.fields["account"])).get(integration_id),
+                "service": dict(get_field_choices(self.fields["service"])).get(service_id),
```

Comment:
rather than introduce `get_field_choices` (sorta abusing choices as a data smuggling mechanism) I would recommend doing it like I did here: 865de3774661a5e443c5673369be4dffa7534d98